Create SECURITY.md (#11485)
kata-containers/kata-containers: SECURITY.md

# Security Policy

Kata Containers is a **rolling-release** project: every monthly release replaces the previous one, and only the _current_ release series receives security fixes. There are **no long-term-support branches**.

---

## Reporting a Vulnerability

### How to report

- **Keep it private first.**
  Please **do not** open a public GitHub issue or pull request for security problems.

- **Preferred: Use GitHub’s security advisory workflow.**
  Follow the official GitHub guide:

>   [Creating a repository security advisory](https://docs.github.com/en/code-security/security-advisories/working-with-repository-security-advisories/creating-a-repository-security-advisory#creating-a-security-advisory)

Reviewer: ```suggestion
  [Report a vulnerability privately](https://docs.github.com/en/code-security/how-tos/report-and-fix-vulnerabilities/report-privately)
```

Reviewer: The original link was pointing to the process of creating an advisory, which is part of the process to fix the vulnerability. In line with that, only a set of roles have access to it. I added the link that points to reporting a vulnerability privately.

- **Alternative (if you don’t have a GitHub account):**

>   Email security concerns to the Kata Containers security team at: **<security@katacontainers.io>**

Reviewer: Does the community already have this alias? Or is the idea to have one created?

This should also refer to the Vulnerability Management Team (VMT), since Kata doesn't have Security Team.

Reviewer 2: No - I remembered Zvonko had discussions with you about this last year. If there isn't one then I'll just drop this option completely.

| **CVE assignment** | Before public disclosure | GitHub automatically requests a CVE ID when we publish the security advisory. |

---

## Supported Versions

| Release | First published | Security-fix window |
|---------|-----------------|---------------------|
| **Latest monthly release** | see `git tag --sort=-creatordate \| head -1` | Actively maintained |
| Any prior release | — | **Unsupported** – please upgrade |

> **Why no backports?**
> Kata’s architecture evolves quickly; back-porting patches would re-introduce the very maintenance burden we avoid by using a rolling model.

### For Downstream Distributions & Vendors

If you maintain a downstream distribution or integration of Kata Containers:

- **Early notification**: Contact the Kata Containers security team to be added to our downstream stakeholders list. We provide advance notification of embargoed security fixes to allow time for rebasing, testing, and release planning.
- **Embargo process**: Before public disclosure, we coordinate appropriate embargo periods to allow your team to incorporate the fix.

> - **Mailing list**: Contact the community manager to subscribe to the Kata Containers security notifications embargo-notice list for advance security patch notification.

Reviewer: ```suggestion
If you maintain a downstream distribution or integration of Kata Containers:

- **Embargo process**: Please be aware that security vulnerabilities are fixed in private as part of an embargo period. During this period only the Kata Containers Vulnerability Management Team (VMT) and very limited number of trusted contributors have access to the vulnerability report and participate in fixing the issue.
- **Early notification**: The Kata Containers VMT sends embargo notifications to the private embargo-notice mailing list a few days in advance of public announcements, to allow downstream stakeholders time to test the fix and prepare to move to the new release that contains the public fix.
- **Subscribing to the mailing list**: Please contact the VMT if you would like to be added to the mailing list.
```

Reviewer 2: The VMT can't add people to the list?

Reviewer 2: I'll just delete this for now if we can't offer clear guidance

---

> ## Disclosure Process & Fix Delivery

Reviewer: This sections seems like a mix of process description for the VMT that contains information for people who reported vulnerabilities or just curious about the process. I think it would be better to point people to the VMT docs. The following section already directs folks to where they can find published advisories, etc.

Reviewer 2: Ok, I will delete this section too

Reviewer: Are you still planning to delete this one?

Reviewer 2: Sorry, I must have missed/ reverted this. Hopefully it's done now?

1. We develop the fix on a private branch.
2. Once validated, we coordinate embargo dates with downstream consumers to allow time for rebasing and testing.
3. We publish a GitHub security advisory, which automatically triggers CVE ID assignment.
4. The fix ships **most commonly** in the next regular monthly release (e.g., `v3.19`) when impact is moderate and waiting does not materially increase risk.
5. **Exception**: For Critical or High-severity issues, we may cut a point release (e.g., `v3.18.1`) to provide an update path for users still on the current series.
6. Upon public release, security details and CVE information are published in the GitHub release notes.

---

## Security Advisories & Release Notes

- Each patch or monthly release includes a **Security Bulletin** section in its GitHub _Release Notes_ summarizing:
  - affected components & versions,
  - CVE identifiers (if assigned),
  - severity / CVSS score,
  - mitigation steps,
  - upgrade instructions.

- We do **not** publish separate “stable-branch” advisories because unsupported branches receive no fixes.

> 

Reviewer: ```suggestion

- To see the list of published security advisories please visit the [security tab](https://github.com/kata-containers/kata-containers/security) in the kata-containers repository.

```

---

## Frequently Asked Questions

#### Q: I run `v3.24` – will you patch it?

A: No. Upgrade to the latest monthly release. Only the current monthly release receives security fixes.

#### Q: I maintain a downstream distribution. Can I get early access to security patches?

A: Yes. Contact the security team (see [SECURITY_CONTACTS](SECURITY_CONTACTS)) to be added to our downstream stakeholders list. We provide embargo periods to allow time for rebasing and testing before public disclosure.

#### Q: What if I don't have a GitHub account? How do I report a vulnerability?

> A: Email <security@katacontainers.io> with a description of the issue. Include steps to reproduce if possible.

Reviewer: Same as above.

Reviewer 2: Removed

#### Q: Where can I discuss a vulnerability once it is public?

> A: Open/continue a GitHub issue **after** the advisory is published, or use the `#kata-containers` Slack channel with a link to the official security advisory.

Reviewer: ```suggestion
A: Open/continue a GitHub issue **after** the advisory is published, or use the `#general` channel in the [Kata Containers Slack workspace](https://join.slack.com/t/katacontainers/shared_invite/zt-16w1u6usn-sK871qbMxVN8KsCP5Gr56A) with a link to the official security advisory.
```

# Security Policy

Kata Containers is a **rolling-release** project: every monthly release replaces the previous one, and only the _current_ release series receives security fixes. There are **no long-term-support branches**.

---

## Reporting a Vulnerability

> 1. **Private first.**  

Author: Indeed. Updated.

   Do **not** open a public GitHub issue or pull request.

2. **Use the offical Github documentation on how to report a vulnerability.**

>    [Creating a repository security advisory](https://docs.github.com/en/code-security/security-advisories/working-with-repository-security-advisories/creating-a-repository-security-advisory#creating-a-security-advisory)

Reviewer: Kata has a detailed VMT process description in the Community repo: https://github.com/kata-containers/community/blob/main/VMT/VMT.md

It only talks about a mailing list to notify downstream stakeholders: https://github.com/kata-containers/community/blob/main/VMT/VMT.md

As an example, the OpenStack VMT team has a wiki page that displays the list of team members and ways to contact them: https://security.openstack.org/vmt.html
I haven't found anything like this for Kata.

# Security Policy

Kata Containers is a **rolling-release** project: every monthly release replaces the previous one, and only the _current_ release series receives security fixes. There are **no long-term-support branches**.

---

## Reporting a Vulnerability

1. **Private first.**  
   Do **not** open a public GitHub issue or pull request.

2. **Use the repository Security tab.**  

>    • Click **“Security ➜ Report a vulnerability.”**  

Author: Done, PTAL

   • This creates a private, access-restricted issue visible only to Kata maintainers and designated security champions.

3. **Response targets (OpenSSF guidelines).**  
   | Action | Target time | Notes |
   | ------ | ----------- | ----- |
   | Initial maintainer response | **≤ 14 calendar days** | Acknowledge receipt and begin triage. |
   | Triage & severity scoring | **≤ 30 days** | We follow CVSS v3.1. |
   | Fix availability | **Next scheduled monthly release**<br/>(or an out-of-band patch release for Critical/High issues) | We may cut `vX.Y.Z` if waiting a full month poses undue risk. |

---

## Supported Versions

| Release | First published | Security-fix window |
|---------|-----------------|---------------------|

> | **Latest monthly release** | see `git tag -l` | Actively maintained |

Reviewer 2: Would `git tag --sort=-creatordate | head -n 1` be a more helpful command here?

| Any prior release | — | **Unsupported** – please upgrade |

> **Why no backports?**  
> Kata’s architecture evolves quickly; back-porting patches would re-introduce the very maintenance burden we avoid by using a rolling model.

---

## Disclosure Process & Fix Delivery

1. We develop the fix on a private branch.  
2. Once validated, we coordinate embargo dates with downstream consumers when appropriate.  
3. The fix ships in **either**:

>    * A point release (e.g., `v3.18.1`) if the vulnerability affects only the current series, **or**

Reviewer 2: Maybe this is nit=picking, but I guess we'd likely move the next release up rather than a patch release as unless it came right after the previous release we'd have new function is, e.g. if we had to release a security fix today I think it would just be 3.19.0?

   * The next regular monthly release (e.g., `v3.19`) when impact is moderate and waiting does not materially increase risk.
4. After the fix is public, we request a CVE ID (if not already issued) and publish details.

---

## Security Advisories & Release Notes

* Each patch or monthly release includes a **Security Bulletin** section in its GitHub *Release Notes* summarizing:
  * affected components & versions,
  * CVE identifiers (if assigned),
  * severity / CVSS score,
  * mitigation steps,
  * upgrade instructions.

* We do **not** publish separate “stable-branch” advisories because unsupported branches receive no fixes.

---

## Frequently Asked Questions

**Q: I run `v3.16` – will you patch it?**  
A: No. Upgrade to the latest monthly release.

**Q: Can I get early access to embargoed fixes?**  

> A: Only project members under the disclosure agreement (see `SECURITY_CONTACTS`) receive advance patches.

Author: I am prepping the dependent PR right now.

> A: Only project members under the disclosure agreement (see `SECURITY_CONTACTS`) receive advance patches.

Author: https://github.com/kata-containers/kata-containers/pull/11486

> A: Only project members under the disclosure agreement (see [SECURITY_CONTACTS](https://kata-containers/kata-containers/SECURITY_CONTACTS)) receive advance patches.

Reviewer 3: Since this is a link to a repo internal file, it can be shortened to :
```suggestion
A: Only project members under the disclosure agreement (see [SECURITY_CONTACTS](SECURITY_CONTACTS)) receive advance patches.
```

---

## Disclosure Process & Fix Delivery

1. We develop the fix on a private branch.  
2. Once validated, we coordinate embargo dates with downstream consumers when appropriate.  
3. We request a CVE ID from MITRE (or another CNA) if one hasn't already been assigned.
4. The fix and vulnerability details are published together in **either**:
   * Common: The next regular monthly release (e.g., `v3.19`) when impact is moderate and waiting does not materially increase risk, **or**
   * Exception: A point release (e.g., `v3.18.1`) if the vulnerability affects only the current series.

---

## Security Advisories & Release Notes

* Each patch or monthly release includes a **Security Bulletin** section in its GitHub *Release Notes* summarizing:
  * affected components & versions,
  * CVE identifiers (if assigned),
  * severity / CVSS score,
  * mitigation steps,
  * upgrade instructions.

* We do **not** publish separate “stable-branch” advisories because unsupported branches receive no fixes.

---

## Frequently Asked Questions

**Q: I run `v3.16` – will you patch it?**  
A: No. Upgrade to the latest monthly release.

**Q: Can I get early access to embargoed fixes?**  

> A: Only project members under the disclosure agreement (see [SECURITY_CONTACTS](SECURITY_CONTACTS)) receive advance patches.

Reviewer 4: Scenario: I'm a vendor, shipping a modified downstream version of Kata with my product. Most of the time, it would not be possible for me to rebase onto the next release or backport a security fix on short notice. Even if I'm notified in advance about an upcoming, _undisclosed_ security fix, I'd still need to integrate and validate it into my downstream branch after it was published.

According to this rule, my product is vulnerable for an extended period of time, and I need to scramble to adopt the patch before its exploited. Is that intended? Could we instead have a process for advance notifications, like https://github.com/kubernetes/committee-security-response/blob/main/private-distributors-list.md?

@kata-containers/architecture-committee

Author: That is a good point. According to this link there is a special mailing list with ditributors that can have access to early fixes that cannot realy on a public release.

- **Alternative (if you don’t have a GitHub account):**
  Email security concerns to the Kata Containers security team at: **security@katacontainers.io**

### What happens after you submit

We follow the OpenSSF vulnerability-handling guidelines.
The table below shows the target timelines we hold ourselves to once we receive your report.
These are independent objectives, not sequential steps.

| Objective | Target time | Notes |
|-----------|-------------|-------|
| **Initial acknowledgement** | ≤ 14 calendar days | Maintainers confirm receipt and start triage. |
| **Triage & CVSS-v3.1 scoring** | ≤ 30 days | We assign severity and plan remediation. |
| **Fix availability** | Next scheduled monthly release<br />(or an out-of-band patch for Critical/High issues) | We may cut a `vX.Y.Z` patch if waiting a month poses undue risk. |
| **CVE assignment** | Before public disclosure | GitHub automatically requests a CVE ID when we publish the security advisory. |

---

## Supported Versions

| Release | First published | Security-fix window |
|---------|-----------------|---------------------|

> | **Latest monthly release** | see `git tag --sort=-creatordate \| head -1` | Actively maintained |

Reviewer 2: The rendering looks okay to me on github

| Any prior release | — | **Unsupported** – please upgrade |

> **Why no backports?**
> Kata’s architecture evolves quickly; back-porting patches would re-introduce the very maintenance burden we avoid by using a rolling model.

### For Downstream Distributions & Vendors

If you maintain a downstream distribution or integration of Kata Containers:

- **Early notification**: Contact the Kata Containers security team to be added to our downstream stakeholders list. We provide advance notification of embargoed security fixes to allow time for rebasing, testing, and release planning.
- **Embargo process**: Before public disclosure, we coordinate appropriate embargo periods to allow your team to incorporate the fix.

> - **Mailing list**: Subscribe to the Kata Containers security notifications list (to be announced) for advance security patch notification.

Reviewer 4: We might as well remove that bullet unless/until we have that list.

Reviewer 2: I'm not sure if this is just the embargo list, which does exist, but we can update that I guess

Reviewer 2: I've updated it to talk to the community manager as that seems to be the current way to do this.

---

## Disclosure Process & Fix Delivery

1. We develop the fix on a private branch.
2. Once validated, we coordinate embargo dates with downstream consumers to allow time for rebasing and testing.
3. We publish a GitHub security advisory, which automatically triggers CVE ID assignment.
4. The fix ships **most commonly** in the next regular monthly release (e.g., `v3.19`) when impact is moderate and waiting does not materially increase risk.
5. **Exception**: For Critical or High-severity issues, we may cut a point release (e.g., `v3.18.1`) to provide an update path for users still on the current series.
6. Upon public release, security details and CVE information are published in the GitHub release notes.

---

## Security Advisories & Release Notes

* Each patch or monthly release includes a **Security Bulletin** section in its GitHub *Release Notes* summarizing:
  * affected components & versions,
  * CVE identifiers (if assigned),
  * severity / CVSS score,
  * mitigation steps,
  * upgrade instructions.

* We do **not** publish separate “stable-branch” advisories because unsupported branches receive no fixes.

---

## Frequently Asked Questions

> **Q: I run `v3.24` – will you patch it?**

Reviewer 4: Consider making the questions micro-headlines (`####` or so), in order to make them linkable.

1. **Private first.**  
   Do **not** open a public GitHub issue or pull request.

> 2. **Use the offical Github documentation on how to report a vulnerabiilty.**

Reviewer 3: Typo. s/vulnerabiilty/vulnerabilty/
